bliss/base_detect/base_common.py

```diff
@@ -0,0 +1,182 @@
+import warnings
+
+import torch
+
+
+def autopad(k, p=None, d=1):  # kernel, padding, dilation
+    # Pad to 'same' shape outputs
+    if d > 1:
+        k = (
+            d * (k - 1) + 1 if isinstance(k, int) else [d * (x - 1) + 1 for x in k]
+        )  # actual kernel-size
+    if p is None:
+        p = k // 2 if isinstance(k, int) else [x // 2 for x in k]  # auto-pad
+    return p
+
+
+class Conv(torch.nn.Module):
+    """Standard convolution."""
+
+    default_act = torch.nn.SiLU()  # default activation
+
+    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, d=1, act=True):
+        """Wrapper for torch.nn.conv2d layer.
+
+        Args:
+            c1: input channels
+            c2: output channels
+            k: kernel size
+            s: kernel stride length
+            p: input padding amount
+            g: controls connections between inputs and outputs
+            d: kernel dilation value
+            act: torch.nn activation function
+        """
+        super().__init__()
+        self.conv = torch.nn.Conv2d(
+            c1, c2, k, s, autopad(k, p, d), groups=g, dilation=d, bias=False
+        )
+        self.bn = torch.nn.BatchNorm2d(c2)
+        self.act = (
+            self.default_act
+            if act is True
+            else act  # noqa: WPS509
+            if isinstance(act, torch.nn.Module)
+            else torch.nn.Identity()
+        )
+
+    def forward(self, x):
+        return self.act(self.bn(self.conv(x)))
+
+    def forward_fuse(self, x):
+        return self.act(self.conv(x))
+
+
+class Conv3d(torch.nn.Module):
+    """Wrapper for torch.nn.conv3d layer.
+
+    Args:
+        c1: input channels
+        c2: output channels
+        k: kernel size
+        s: kernel stride length
+        p: input padding amount
+        g: controls connections between inputs and outputs
+        d: kernel dilation value
+        act: torch.nn activation function
+    """
+
+    default_act = torch.nn.SiLU()  # default activation
+
+    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, d=1, act=True):
+        super().__init__()
+        self.conv = torch.nn.Conv3d(
+            c1, c2, k, s, autopad(k, p, d), groups=g, dilation=d, bias=False
+        )
+        self.bn = torch.nn.BatchNorm3d(c2)
+        self.act = (
+            self.default_act
+            if act is True
+            else act  # noqa: WPS509
+            if isinstance(act, torch.nn.Module)
+            else torch.nn.Identity()
+        )
+
+    def forward(self, x):
+        return self.act(self.bn(self.conv(x)))
+
+    def forward_fuse(self, x):
+        return self.act(self.conv(x))
+
+
+class Conv3d_down(Conv3d):  # noqa: N801
+    def forward(self, x):
+        z = self.act(self.bn(self.conv(x)))
+        return torch.squeeze(z, 1)
+
+
+class Bottleneck(torch.nn.Module):
```

Comment:
Why do we need to include `Bottleneck` in our repo, rather than using the version in yolov5?

Reply:
Same for some of the other classes below.

Reply:
Everything in `base_detect` is outdated. We can delete that entire directory now